django_cloud_tasks/tasks/task.py

```diff
@@ -319,18 +319,23 @@ def discard(cls, task_id: str | None = None, min_retries: int = 0):
         else:
             task_objects = client.list_tasks(queue_name=cls.queue())
 
-        outputs = []
-        for task_obj in task_objects:
+        def process_task(task_obj, min_retries, client):
+            outputs = []
             task_name = task_obj.http_request.url.rsplit("/", 1)[-1]
-            if task_name != cls.name():
-                continue
-
-            if task_obj.dispatch_count < min_retries:
-                continue
+            if task_name == cls.name() and task_obj.dispatch_count >= min_retries:
+                task_id = task_obj.name.split("/")[-1]
+                client.delete_task(queue_name=cls.queue(), task_name=task_id)
+                outputs.append(f"{task_name}/{task_id}")
+            return outputs
 
-            task_id = task_obj.name.split("/")[-1]
-            client.delete_task(queue_name=cls.queue(), task_name=task_id)
-            outputs.append(f"{task_name}/{task_id}")
+        outputs = []
+        with ThreadPoolExecutor() as executor:
+            future_to_task = {
+                executor.submit(process_task, task_obj, min_retries, client): task_obj for task_obj in task_objects
+            }
+            for future in as_completed(future_to_task):
+                task_outputs = future.result()
+                outputs.extend(task_outputs)
```

Comment:
A design that I find very readable is defining _what to do_ (function) and _do with what_ (items) by using map function:

```
def process(task_obj):
    task_id = task_obj.name.split("/")[-1]
    client.delete_task(queue_name=cls.queue(), task_name=task_id)
    return f"{task_name}/{task_id}"
    
pool = ThreadPoolExecutor(max_workers=max_workers)

outputs = []
for output in pool.map(process, task_objects):
    outputs.append(output)
return outputs
```

Reply:
I believe this code is more 'Pythonic.' I'm going to test if it still works